extensions/guacamole-auth-totp/src/main/java/org/apache/guacamole/auth/totp/user/UserVerificationService.java

```diff
@@ -108,7 +108,7 @@ private UserTOTPKey getKey(UserContext context,
 
         // If no key is defined, attempt to generate a new key
         String secret = attributes.get(TOTPUser.TOTP_KEY_SECRET_ATTRIBUTE_NAME);
-        if (secret == null) {
+        if (secret == null || secret.isEmpty()) {
```

Comment:
Why expand the representation of no defined key to include the empty string?

Reply:
I ran into a few situations testing out my changes where the DB value for the secret attribute was an empty string, and it causes problems - basically generates an invalid QR code that caused an error when you try to add it in to something like Google Authenticator.  I can leave it as just `null` if you like, but it seemed like a reasonable check to have in there.